File: MiniDOMBase/DMD/Sistema/Users/CUserComparer.vb
Imports System.Net.Mail 'importo il Namespace
Imports System.Net.Sockets
Imports minidom
Imports minidom.Databases
Imports minidom.Anagrafica

Partial Public Class Sistema

    Public Class CUserComparer
        Implements IComparer

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub

        Public Function Compare(ByVal user1 As CUser, ByVal user2 As CUser) As Integer
            Return user1.CompareTo(user2)
        End Function

        Private Function Compare1(ByVal a As Object, ByVal b As Object) As Integer Implements IComparer.Compare
            Return Me.Compare(a, b)
        End Function

    End Class

End Class

File: MiniDOMBase/DMD/Sistema/Users/Comparers/CUserComparer.cs
using System.Collections;
using System.Collections.Generic;

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Oggetto IComparer
        /// </summary>
        public class CUserComparer 
            : IComparer, IComparer<CUser>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CUserComparer()
            {
                //DMDObject.IncreaseCounter(this);
            }

            /// <summary>
            /// Compara due oggetti
            /// </summary>
            /// <param name="user1"></param>
            /// <param name="user2"></param>
            /// <returns></returns>
            public int Compare(CUser user1, CUser user2)
            {
                return user1.CompareTo(user2);
            }

            /// <summary>
            /// Compara due oggetti
            /// </summary>
            /// <param name="a"></param>
            /// <param name="b"></param>
            /// <returns></returns>
            int IComparer.Compare(object a, object b)
            {
                return Compare((CUser)a, (CUser)b);
            }
        }
    }
}